File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.java
class Solution {
    public String solution(int n, int t, int m, int p) {
        String answer = "";
        String tmp = "";
        for (int i = 0; i <= (m * t); i++) {
            tmp += Integer.toString(i, n);
        }

        
        for (int i = 0; i < t; i++) {
            char c = tmp.charAt(p-1 + (m*i));
            if (!Character.isDigit(c)) {
                answer += Character.toUpperCase(c);
            } else {
                answer += c;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.py
def solution(n, t, m, p):
    answer = ""
    tmp = ""
    
    num = 0
    while len(tmp) < t * m:
        tmp += convert_to_base(num, n)
        num += 1
        
    for i in range(t):
        answer += tmp[p-1 + (m*i)]
    
    return answer

def convert_to_base(num, base):
    digits = "0123456789ABCDEF"
    if (num == 0):
        return "0"
    
    result = ""
    while num > 0:
        result = digits[num % base] + result
        num //= base
    
    return result